src/Infrastructure/Event/EventHistory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Infrastructure\Event;
+
+use App\Domain\Event\EventHistoryInterface;
+use Doctrine\DBAL\Driver\Connection;
+use Ramsey\Uuid\Uuid;
+
+class EventHistory implements EventHistoryInterface
+{
+    /**
+     * @return \Ramsey\Uuid\UuidInterface[]|[]
+     */
+    public function getStreamIds(string $filter = null)
+    {
+        $query = 'SELECT DISTINCT `uuid` FROM ' . $this->eventStoreTableName;
+        if ($filter) {
+            $query .= ' WHERE `type` = :type';
```

Comment:
In theory what you replay it's an event stream (Aggregate Root), entire or partially, from revision to revision. But I don't get any use case when replaying only a type of events. Can you give me one?